app/user.go

```diff
@@ -1853,3 +1873,87 @@ func (a *App) UpdateOAuthUserAttrs(userData io.Reader, user *model.User, provide
 
 	return nil
 }
+
+// UsersCanJoinTeam verifies that users can join a team based on group-constraints. If one or more users cannot join
+// the team a model.InvalidTeamMembersError is returned containing the user ids of the denied users.
+func (a *App) UsersCanJoinTeam(userIDs []string, team *model.Team) error {
```

Comment:
As a developer reading the API Name, it is not clear to me that this does not enforce other permission / type checks, e.g. is it a private channel? - Reading the Method name would suggest that to me. Similar with the other `*PermittedTo*` functions. Should we make it clear via the name that this is only checking for Group Constraints?

Reply:
This is a good point that wasn't on my radar. 

There are (currently un-editable) permissions to join teams and channels that these methods intentionally don't consider. Also they intentionally don't consider email domain restrictions or, in the case of joining a channel, whether the user is a member of the parent team.

All of the above points could make the current name confusing as you rightly pointed out. I'm going to think of and propose a rename.